src/main/java/com/internet/shop/servise/impl/ShoppingCartServiceImpl.java

```diff
@@ -1,7 +1,56 @@
 package com.internet.shop.servise.impl;
 
+import com.internet.shop.dao.interfaces.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Inject;
 import com.internet.shop.lib.Service;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.servise.interfaces.ShoppingCartService;
+import java.util.ArrayList;
 
 @Service
-public class ShoppingCartServiceImpl {
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCartDao.get(shoppingCart.getId()).get()
+                .getProducts()
+                .add(product);
+        return shoppingCartDao.get(shoppingCart.getId()).get();
```

Comment:
```suggestion
        shoppingCart.getProducts().add(product);
        return shoppingCartDao.update(shoppingCart);
```
подібно зробити в методі deleteProduct і clear()